server/src/main/java/org/candlepin/dto/api/APIDTOFactory.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api;
+
+import org.candlepin.dto.DTOFactory;
+
+
+
+/**
+ * The APIDTOFactory is an abstract class which denotes factories which are pre-configured with
+ * translators for creating DTOs compatible with the most recent API specification.
+ */
+public abstract class APIDTOFactory extends DTOFactory {
```

Comment:
Maybe 

* DTOFactory - interface
* BaseDTOFactory - class implements DTOFactory
* APIDTOFactory - interface
* APIv1DTOFactory - extends BaseDTOFactory and implements APIDTOFactory